test/functional/apps/discover/context_awareness/_data_source_profile.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import kbnRison from '@kbn/rison';
+import expect from '@kbn/expect';
+import type { FtrProviderContext } from '../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const PageObjects = getPageObjects(['common', 'timePicker', 'discover', 'unifiedFieldList']);
+  const testSubjects = getService('testSubjects');
+  const dataViews = getService('dataViews');
+
+  describe('data source profile', () => {
+    describe('ES|QL mode', () => {
+      describe('cell renderers', () => {
+        it('should render custom @timestamp but not custom log.level', async () => {
+          const state = kbnRison.encode({
+            dataSource: { type: 'esql' },
+            query: { esql: 'from my-example-* | sort @timestamp desc' },
```

Comment:
I have question for my clarity. Even though `my-example-*` includes `my-example-logs` as an index, why `log.level` is not customized here?

Reply:
Good question, it's just due to the way the resolution logic for the example profile that customizes `log.level` is implemented here: https://github.com/elastic/kibana/blob/07a26a82723d877ea2e469ea3f411e328ffa662e/src/plugins/discover/public/context_awareness/profile_providers/example_data_source_profile/profile.tsx#L62-L64

We're only matching on this profile if the index pattern exactly equals `my-example-logs`, although it could have been implemented to match on any index pattern covered by `my-example-*` if I had decided to implement it that way. Real resolution implementations like the logs one are more sophisticated, this one is pretty dumb just for example purposes.

Reply:
Ahh... I went though these example profile multiple time but missed this line. Sorry for that and thank you very much for pointing it out.

>Real resolution implementations like the logs one are more sophisticated, this one is pretty dumb just for example purposes.

Makes sense to me now. This demonstrates the profile resolutions well enough. Thanks.